legacy-migration/providers/migrate.py

```diff
@@ -0,0 +1,312 @@
+        contact_person["Roles"] = [DEFAULT_CONTACT_ROLE]
+        contact_person['LastName'] = contact['last_name']
+        # TODO adding first name
+        contact_person['FirstName'] = contact['first_name']
+
+    # The old-style provider record contains only a single address value
+        if contact.get("address"):
+            address = parse_addresses(contact)
+            addresses.append(address)
+
+        if contact.get("email"):
+            email_value = parse_emails(contact)
+            email_contact = {"Type": "Email", "Value": email_value}
+            contact_mechanisms.append(email_contact)
+
+        if contact.get("phones"):
+            phone_numbers = parse_phones(contact)
+            for phone_number in phone_numbers:
+                phone_contact = {"Type": "Telephone", "Value": phone_number}
+                # Use the same contact mechanism object as emails
+                contact_mechanisms.append(phone_contact)
+
+        if len(addresses) > 0 or len(contact_mechanisms) > 0 or len(phone_contacts) > 0 :
+            contact_person['ContactInformation'] = {}
+
+        # Addresses are optional if there were any, add them
+        if len(addresses) > 0:
+            # contact_person['ContactInformation'] = {}
+            # We must wrap the contact person with contract information
+            contact_person['ContactInformation']['Addresses'] = addresses
+
+        # If emails or phones were provided add them to the metadata
+        if len(contact_mechanisms) > 0:
+            contact_person['ContactInformation']['ContactMechanisms'] = contact_mechanisms
+
+        # Add to contact person to contact_persons arr
+        contact_persons.append(contact_person)
+        return contact_persons
+        #TODO we need to handle if there are more fields provided
+
+def migrate_organizations(search_urls, provider_id,org_name):
+    """Parses organization's fields between old-style format into new schema"""
+    organizations = []
+    for url in search_urls:
+        organization = {}
+        # In all cases on PROD it was found that
+        # the provider_id and Shortname were always the same this
+        # will be enforced in the new schema
+        organization["ShortName"] = provider_id
+        organization["LongName"] = org_name
+        organization["URLValue"] = url
+        # Consider passing a default argument for the organization roles
+        organization["Roles"] = [DEFAULT_ORGANIZATION_ROLE]
+        organizations.append(organization)
+    return organizations
+
+def migrate_administrators(provider_owner_guid, provider_id):
+    """Parses admin's fields between old-style format into new schema"""
+    all_the_groups_specific_provider = get_groups_by_provider(provider_owner_guid)
+    member_guids_specific_provider = get_all_provider_guids(all_the_groups_specific_provider)
+    # TODO: Iron out the pieces
+    if member_guids_specific_provider is None:
```

Comment:
so maybe an else statement to print out all the found admins to a log file, I'm just now thinking it would be nice to have an output of all the people who have been selected to be an admin across the entire holdings. This would then be used to verify that we actually know all these people and the accounts are active.